community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactions.java

```diff
@@ -219,13 +226,15 @@ public boolean haveClosingTransaction()
     @Override
     public void start() throws Throwable
     {
+        stopped = false;
         unblockNewTransactions();
     }
 
     @Override
     public void stop() throws Throwable
     {
         blockNewTransactions();
+        stopped = true;
```

Comment:
newTransactionsLock and assertCurrentThreadIsNotBlockingNewTransactions is not relevant for this kind of case because this assertion will not gonna help you to terminate threads that want to start new transaction, while KernelTransactions are stopped (it can be the case that this particular KernelTransactions instance will not gonna be started ever again and we want to release those threads in that case), in the same time we do not want transactions that are started during safe zone marking phase to be terminated for non related reason.